packages/api/src/external/carequality/organization-conversion.ts

```diff
@@ -3,11 +3,14 @@ import { XCPDGateways } from "@metriport/ihe-gateway-sdk";
 import { CQOrgBasicDetails } from "./command/cq-directory/search-cq-directory";
 
 export function cqOrgsToXCPDGateways(cqOrgs: CQOrgBasicDetails[]): XCPDGateways {
-  return cqOrgs.map(org => {
-    return {
-      id: uuidv7(),
-      oid: org.id,
-      url: org.urlXCPD,
-    };
+  return cqOrgs.flatMap(org => {
+    if (org.urlXCPD) {
+      return {
+        url: org.urlXCPD,
+        oid: org.id,
+        id: uuidv7(),
```

Comment:
if this uuid is random and is generated per request, it should be generated in the logic of `prepareForPatientDiscovery` or `createPatientDiscoveryRequest` and not here in a more generic conversion function